File: server-status-tracker/src/main/java/com/server/status/tacker/payload/response/ServerResponse.java
package com.server.status.tacker.payload.response;

import com.server.status.tacker.enumeration.Status;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * @Created 25/06/2024 - 09:58
 * @Package com.server.status.tacker.payload.response
 * @Project server-status-tracker
 * @User LegendDZ
 * @Author Abdelaaziz Ouakala
 **/
@Data
public class ServerResponse {
    private Long id;
    private String uid;
    private String owner;
    private String ipAddress;
    private String name;
    private String memory;
    private String icon;
    private String type;
    private Status status;
    private LocalDateTime lastCheck;
    private String lastCheckTimeAgo;
    private LocalDateTime createdAt;
    private String createdAtTimeAgo;
    private LocalDateTime updatedAt;
    private String updatedAtTimeAgo;
}

File: server-status-tracker-client/src/app/model/server-response.ts
import { Status } from './status';

export interface ServerResponse {
  id: number;
  uid: string;
  name: string;
  owner: string;
  ipAddress: string;
  memory: string;
  type: string;
  icon: string;
  status: Status;
  lastCheck: Date;
  lastCheckTimeAgo: string;
  createdAt: Date;
  createdAtTimeAgo: string;
  updatedAt: Date;
  updatedAtTimeAgo: string;
}
